File: LeetCode/LeetCode Practice Problems/1684- Count the Number of Consistent Strings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        set<char> s;
        
        for(char c: allowed) {
            s.insert(c);
        }
        
        int ans = 0;
        for(string a:words) {
            bool flag = true;
            for(char c: a) {
                if(s.find(c) == s.end()) {
                    flag = false;
                    break;
                }
            }
            
            if(flag) ans++;
        }
        
        return ans;
    }
};

File: LeetCode/LeetCode Practice Problems/1684-Count the Number of Consistent Strings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        int cnt[] = new int[26];
        
        for(char c: allowed.toCharArray()) {
            cnt[c - 'a']++;
        }
        
        int ans = 0;
        
        for(String word: words) {
            int i = 0;
            
            while(i < word.length() && cnt[word.charAt(i) - 'a'] > 0) {
                i++;
            }
            
            if(i == word.length()) {
                ans++;
            }
        }
        
        return ans;
    }
}
